Public/Src/Engine/UnitTests/Scheduler/PipFilterTest.cs

```diff
@@ -407,7 +407,30 @@ public void TestRequiredInputsFilter()
             AssertSetEquals(expectedDependencyOutputs, actualDependentsOutputs);
         }
 
-        private void SetupDependenciesFilterGraph(out FileArtifact o0, out FileArtifact o1, out FileArtifact o2, out Process p2, out PipGraph graph)
+        [Fact]
+        public void TestMultipleRequiredInputsFilter()
+        {
+            // P0 -> P1 -> P2 -> Seal -> P3 (P3 doesn't matter in this test case, just to reuse the existing SetupDependenciesFilterGraph funcion)
+            FileArtifact o0, o1, o2;
+            Process p1, p2;
+            PipGraph graph;
+            SetupDependenciesFilterGraph(out o0, out o1, out o2, out p1, out p2, out graph);
+
+            var filter =
+                new RootFilter(
+                    new DependenciesFilter(

```

Comment:
>DependenciesFilter [](start = 24, length = 18)

I don't see any other tests for DependenciesFilter operating on a set of more than 1 pip. Could you add test cases to cover those scenarios where the inner filter yields a nonzero set of pips but the ClosureMode is TransitiveIncludingSelf?

Reply:
Made the test case a theory and add the TransitiveIncludingSelf to it.